### ISSUE TITLE
"Sequence contains no elements" error

### ISSUE THREAD

**Issue by LouisMilesRH** [dotnet/Scaffolding#1708]
### Description
Having updated my project to .NET 6 I am no longer able to use the Scaffolding tool built into Visual Studio (having tried both VS 2019 and VS 2022) to generate a controller for my very simple model containing only an ID and Name field. The following error message is returned...

"Sequence contains no elements"

![image](https://user-images.githubusercontent.com/59727334/142409664-3ad593ca-7f52-4f9a-a72d-29c56ba1f47f.png)

When downgrading the project to .NET 5 (and simultaniously downgrading required nuget packages including Microsoft.VisualStudioWeb.CodeGeneration.Design to their .NET 5 versions) the model is correctly scaffolded leading me to believe it must be an error in the latest .NET 6 version.

**Comment by Rick-Anderson** [comment]
use the [Report a Problem](https://docs.microsoft.com/visualstudio/ide/how-to-report-a-problem-with-visual-studio) gesture  to report this. However, many of us have upgraded to .NET 6 are are successfully using scaffolding. For sure VS 2019 won't work with .NET 6/scaffolding.

**Comment by abdulqadir89** [comment]
Facing same problem. Scaffolding works fine with new mvc project in VS 2022. There must be some config changes which we are missing.

**Comment by LouisMilesRH** [comment]
Sorry I probably should have replied back to this as I did eventually get it working...

In order to fix it I needed to change my Program.cs to the hosting model style as mentioned here https://docs.microsoft.com/en-us/aspnet/core/migration/50-to-60?view=aspnetcore-6.0&tabs=visual-studio#new-hosting-model

Note that the app itself worked quite happily in the older method but changing to use scaffolding I needed to change this over. Will leave up to the developers if they want to close this or treat as a bug but hopefully others find it helpful.

**Comment by Rick-Anderson** [comment]
@LouisMilesRH  can you close this issue?

**Comment by abdulqadir89** [comment]
@LouisMilesRH Have you changed anything other than Program.cs?

**Comment by abdulqadir89** [comment]
I have changed program.cs as mentioned. Still its not working.

I created a new MVC project and copied just models one by one. Scaffolding worked just for simple models. Once I copied complex models having many to many relationships, it stopped working with this "sequence contains no elements" error. 
After that Scaffolding for simple models having just two properties without any relationship is also not working.
